File: tp3/testing/processo.c
#include <arpa/inet.h>
#include <stdio.h>
#include <string.h>
#include <sys/socket.h>
#include <unistd.h>
#include <stdlib.h>
#include <time.h>

#define PORT 8080

int main(int argc , char *argv[])
{
    int sock = 0;
    char buffer[10] = {0};
    struct sockaddr_in serv_addr;
    if ((sock = socket(AF_INET, SOCK_STREAM, 0)) < 0)
    {
        printf("\n Socket creation error \n");
        return -1;
    }

    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons(PORT);

    if (inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr)
        <= 0)
    {
        printf(
            "\nInvalid address/ Address not supported \n");
        return -1;
    }

    if (connect(sock, (struct sockaddr*)&serv_addr,
                sizeof(serv_addr))
        < 0)
    {
        printf("\nConnection Failed \n");
        return -1;
    }
    
    for(int j = 0; j < 3; j++)
    {   
        char msg[] = "John China";
        send(sock, msg, sizeof(msg), 0);
    }

    return 0;
}

File: tp3/testing/processo.py
import socket
#import sys
import os
from time import sleep

try:
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    print ("Socket successfully created")
except socket.error as err:
    print ("socket creation failed with error %s" %(err))
 
# connecting to the server
s.connect(("127.0.0.1", 8080))

pid = os.getpid()

for i in range(2):
    sleep(2)
    s.send(f"{pid} | {i}\n".encode())
